operator/internal/deploy/chart_manager.go

```diff
@@ -0,0 +1,222 @@
+	remoteClientCache *remote.ClientCache, kcpClient client.Client,
+) (bool, error) {
+	skrClient, err := remote.NewRemoteClient(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy, remoteClientCache)
+	if err != nil {
+		return true, err
+	}
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy)
+	if err != nil {
+		return true, err
+	}
+	argsVals, err := m.generateHelmChartArgs(ctx, kcpClient)
+	if err != nil {
+		return true, err
+	}
+	// TODO(khlifi411): make sure that validating-webhook-config resource is in sync with the secret configuration
+	skrWatcherInstallInfo := prepareInstallInfo(ctx, m.config.WebhookChartPath, ReleaseName, skrCfg, skrClient, argsVals)
+	err = installOrRemoveChartOnSKR(ctx, skrWatcherInstallInfo, ModeInstall, m.config.EnableWebhookPreInstallCheck)
+	if err != nil {
+		return true, err
+	}
+	kyma.UpdateCondition(v1alpha1.ConditionReasonSKRWebhookIsReady, metav1.ConditionTrue)
+	return false, nil
+}
+
+func (m *EnabledSKRWebhookChartManager) RemoveWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+	syncCtx *remote.KymaSynchronizationContext,
+) error {
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, syncCtx.ControlPlaneClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy)
+	if err != nil {
+		return err
+	}
+	argsVals, err := m.generateHelmChartArgs(ctx, syncCtx.ControlPlaneClient)
+	if err != nil {
+		return err
+	}
+	skrWatcherInstallInfo := prepareInstallInfo(ctx, m.config.WebhookChartPath, ReleaseName, skrCfg,
+		syncCtx.RuntimeClient, argsVals)
+	return installOrRemoveChartOnSKR(ctx, skrWatcherInstallInfo, ModeUninstall, m.config.EnableWebhookPreInstallCheck)
+}
+
+func (m *EnabledSKRWebhookChartManager) generateHelmChartArgs(ctx context.Context,
+	kcpClient client.Client,
+) (map[string]interface{}, error) {
+	watcherList := &v1alpha1.WatcherList{}
+	if err := kcpClient.List(ctx, watcherList, &client.ListOptions{}); err != nil {
+		return nil, fmt.Errorf("error listing watcher resources: %w", err)
+	}
+	if len(watcherList.Items) == 0 {
+		return nil, ErrFoundZeroWatchers
+	}
+	chartCfg := generateWatchableConfigs(watcherList)
+	bytes, err := k8syaml.Marshal(chartCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	kcpAddr, err := m.resolveKcpAddr(ctx, kcpClient)
+	if err != nil {
+		return nil, err
+	}
+	return map[string]interface{}{
+		"triggerLabel":          time.Now().Format(triggerLabelTimeFormat),
+		"kcpAddr":               kcpAddr,
+		"resourcesLimitsMemory": m.config.SkrWebhookMemoryLimits,
+		"resourcesLimitsCPU":    m.config.SkrWebhookCPULimits,
+		customConfigKey:         string(bytes),
+	}, nil
+}
+
+func (m *EnabledSKRWebhookChartManager) resolveKcpAddr(ctx context.Context, kcpClient client.Client) (string, error) {
+	if m.kcpAddr != "" {
+		return m.kcpAddr, nil
+	}
+	// Get external IP from the ISTIO load balancer external IP
+	loadBalancerService := &corev1.Service{}
```

Comment:
This implementation is too opinionated on load balancer only. I think we should at least offer node-port to allow implementation locally.

Reply:
As discussed. This is a bigger refactoring which will be done here:
#303

Reply:
THIS is part of the bigger refactoring :D